feature/platform/transceiver/otg_tests/zr_temperature_test/zr_temperature_test.go

```diff
@@ -0,0 +1,225 @@
+			sensorComponent := gnmi.Get[*oc.Component](t, dut1, gnmi.OC().Component(subc.GetName()).State())
+			if sensorComponent.GetType() == sensorType {
+				scomponent := gnmi.OC().Component(sensorComponent.GetName())
+				if scomponent != nil {
+					component1 = scomponent
+				}
+			}
+		}
+	}
+	p1StreamInstant := samplestream.New(t, dut1, component1.Temperature().Instant().State(), 10*time.Second)
+	p1StreamAvg := samplestream.New(t, dut1, component1.Temperature().Avg().State(), 10*time.Second)
+	p1StreamMin := samplestream.New(t, dut1, component1.Temperature().Min().State(), 10*time.Second)
+	p1StreamMax := samplestream.New(t, dut1, component1.Temperature().Max().State(), 10*time.Second)
+	// Wait 120 sec cooling off period
+	gnmi.Await(t, dut1, gnmi.OC().Interface(dp1.Name()).OperStatus().State(), intUpdateTime, oc.Interface_OperStatus_DOWN)
+	temperatureInstant := verifyTemperatureSensorValue(t, dut1, p1StreamInstant, "Instant")
+	t.Logf("Port1 dut1 %s Instant Temperature: %v", dp1.Name(), temperatureInstant)
+	temperatureMax := verifyTemperatureSensorValue(t, dut1, p1StreamMax, "Max")
+	t.Logf("Port1 dut1 %s Max Temperature: %v", dp1.Name(), temperatureMax)
+	temperatureMin := verifyTemperatureSensorValue(t, dut1, p1StreamMin, "Min")
+	t.Logf("Port1 dut1 %s Min Temperature: %v", dp1.Name(), temperatureMin)
+	temperatureAvg := verifyTemperatureSensorValue(t, dut1, p1StreamAvg, "Avg")
+	t.Logf("Port1 dut1 %s Avg Temperature: %v", dp1.Name(), temperatureAvg)
+	i = d.GetOrCreateInterface(dp1.Name())
+	i.Enabled = ygot.Bool(true)
+	i.Type = oc.IETFInterfaces_InterfaceType_ethernetCsmacd
+	// Enable interface
+	gnmi.Replace(t, dut1, gnmi.OC().Interface(dp1.Name()).Config(), i)
+	gnmi.Await(t, dut1, gnmi.OC().Interface(dp1.Name()).OperStatus().State(), intUpdateTime, oc.Interface_OperStatus_UP)
+	temperatureInstant = verifyTemperatureSensorValue(t, dut1, p1StreamInstant, "Instant")
+	t.Logf("Port1 dut1 %s Instant Temperature: %v", dp1.Name(), temperatureInstant)
+	temperatureMax = verifyTemperatureSensorValue(t, dut1, p1StreamMax, "Max")
+	t.Logf("Port1 dut1 %s Max Temperature: %v", dp1.Name(), temperatureMax)
+	temperatureMin = verifyTemperatureSensorValue(t, dut1, p1StreamMin, "Min")
+	t.Logf("Port1 dut1 %s Min Temperature: %v", dp1.Name(), temperatureMin)
+	temperatureAvg = verifyTemperatureSensorValue(t, dut1, p1StreamAvg, "Avg")
+	t.Logf("Port1 dut1 %s Avg Temperature: %v", dp1.Name(), temperatureAvg)
+	if temperatureAvg >= temperatureMin && temperatureAvg <= temperatureMax {
+		t.Logf("The average is between the maximum and minimum values")
+	} else {
+		t.Fatalf("The average is not between the maximum and minimum values")
+	}
+}
+
+func opticalChannelFromPort(t *testing.T, dut *ondatra.DUTDevice, p *ondatra.Port) string {
+	t.Helper()
+	if deviations.MissingPortToOpticalChannelMapping(dut) {
+		switch dut.Vendor() {
+		case ondatra.ARISTA:
+			transceiverName := gnmi.Get(t, dut, gnmi.OC().Interface(p.Name()).Transceiver().State())
```

Comment:
same as above is transceiverName really all of the transceiver state and not the name? If so, this should change to transeiverState?

Reply:
hi Ashsaan,

This return transceiver component name same code snippet suggested in review comment from LimeHat.